### PR TITLE
Allow updating field names (and names and descriptions)

### PR DESCRIPTION
**Pull request by rjmac**
...with plumbing of the field name updates to data-coordinator

### CODE REVIEW

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala [hunk 1]
```diff
@@ -205,6 +205,78 @@ class ColumnDAOImpl(dc: DataCoordinatorClient, store: NameAndSchemaStore, column
     }
   }
 
+  def doUpdateColumn(datasetRecord: DatasetRecord, columnRecord: ColumnRecord, user: String, precondition: Precondition, column: ColumnName, spec: UserProvidedColumnSpec, requestId: RequestId): Result = {
+    log.warn("TODO: updating existing columns not supported yet")
```
Comment (chitang) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala]: This warning is out dated?

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala [hunk 2]
```diff
@@ -205,6 +205,78 @@ class ColumnDAOImpl(dc: DataCoordinatorClient, store: NameAndSchemaStore, column
...
+      if(newDatatype != columnRecord.typ) return ColumnDAO.CannotChangeColumnType
+    }
+    spec.id.foreach { newId =>
+      if(newId != columnRecord.id) return ColumnDAO.CannotChangeColumnId
+    }
+
+    store.withColumnUpdater(datasetRecord.systemId, columnRecord.id) { updater =>
+      val instructions = List.newBuilder[DataCoordinatorInstruction]
+      spec.fieldName.foreach { fn =>
+        updater.updateFieldName(fn)
```
Comment (chitang) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala]: Can we not update if it is hasn't changed?

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala (hunk 1 above)
Comment (rjmac) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala]: Oh, yes you're right.

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala (hunk 2 above)
Comment (rjmac) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/ColumnDAOImpl.scala]: We can, and I considered it, but it's a lot of noise for not a lot of benefit.

**Pull request merged**